### PR TITLE
refactor: break agentskills single file to multiple composable modules

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Prereq of #3139.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- This PR re-organizes a single *huge* `agentskills.py` into multiple sub-components. This makes the module more readable and manageable and will make future integration of Agentless easier.
- This PR also set disable logger's log to file behavior by default - it will cause permission issues if we import `opendevin` inside runtime as the logger will try to create `logs/` which is not super necessary.

---
**Other references**

### CODE REVIEW

File: containers/app/Dockerfile [hunk 1]
```diff
@@ -81,8 +81,6 @@ RUN python opendevin/core/download.py # No-op to download assets
 # opendevin:opendevin -> opendevin:app
 RUN find /app \! -group app -exec chgrp app {} +
 
-RUN chown -R opendevin:app /app/logs && chmod -R 770 /app/logs # This gets created by the download.py script
```
Comment (tobitege) [inline containers/app/Dockerfile:84]: sure about this?

Comment (xingyaoww) [reply containers/app/Dockerfile:84]: Yep - just build the `app` container and tested locally - seems to work!

Comment (tobitege) [reply containers/app/Dockerfile:84]: ok, was surprised seeing a Dockerfile change in this PR, just wanted to make sure :)

File: opendevin/core/logger.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 DISABLE_COLOR_PRINTING = False
 DEBUG = os.getenv('DEBUG', 'False').lower() in ['true', '1', 'yes']
+LOG_TO_FILE = os.getenv('LOG_TO_FILE', 'False').lower() in ['true', '1', 'yes']
```
Comment (enyst) [inline opendevin/core/logger.py:13]: Can you please tell why is this necessary, is the actual issue that when this file is imported, it runs code? That is, if it didn't run its initialization at import, would it work fine?

Comment (xingyaoww) [reply opendevin/core/logger.py:13]: Yes, so whenever `opendevin` is imported, it will try to by default creates `logs/` directory under the project root, which is kinda not necessary for client running inside runtime, and for server running inside the `opendevin app` docker. 

Whenever the project root has a different write permission for the current user, the `mkdir logs/` will fail and break the code. Since imo `logs/` will primarily be accessible and useful during debugging session, we can just set it as an optional behavior, rather than default behavior, which allows us to simplify other part a bit - wdyt?

**Review by tobitege — APPROVED** [review]
From my short local test, this seems to work fine. LGTM

File: opendevin/core/logger.py (hunk 2 above)
Comment (enyst) [reply opendevin/core/logger.py:13]: OK, I'm [refactoring out](https://github.com/OpenDevin/OpenDevin/pull/3413/files#diff-c6567843b80b57fcc1d61101cf9fd68a5fcf75f3f24ae4236d1dce64b4531f5fR286) the initialization during import, and also plan to make the log directory for those llm logs configurable. Maybe it will also address this? To try to replicate it, I should run the `app` docker with a different user then, that's the only way it happens?

I don't know about removing logs by default. We sometimes ask for logs from users to figure out what happened, and people are also asking to see LLM prompts. But maybe I'm missing something, I didn't try it yet!

Comment (enyst) [reply opendevin/core/logger.py:13]: I'm also thinking to split the runtime log from the general `opendevin.log`, so what comes from docker basically, to be in its own place. Currently the general file has everything except llm prompts, it's a lot, specially now with the new runtime. What do you think?

Comment (tobitege) [reply opendevin/core/logger.py:13]: That could complicate things a bit to keep an eye on both, so to speak. The docker logs can be a bit spammy, but especially in case of issues, they're worth it and pretty "convenient" to have them in the same log _at the right chronological order_.

Comment (tobitege) [reply opendevin/core/logger.py:13]: Maybe we could reduce the amount of Docker log lines, though, by only including the newest lines since the last log grab (and not always "the last 10")? 🤔

Comment (enyst) [reply opendevin/core/logger.py:13]: This is only the files, they are still in console. Although... wait, no, not split the runtime log, but make it possible to enable/disable or set to debug separately from the rest.

Comment (tobitege) [reply opendevin/core/logger.py:13]: Yes, that sounds good

Comment (xingyaoww) [reply opendevin/core/logger.py:13]: yeah, i think overall have a good way to control when/how we can log is the ideal way! I'd still hope the logging is by default false (e.g., when people import opendevin, they won't expect opendevin will create a new folder in their disk). We can add env var to enable this in the docker, if we really want this behavior -- but the issue is, even if that's enabled, the user will have hard-time accessing those logs, since they are stored *inside* the docker container.
